middleware.js

```diff
@@ -202,31 +39,32 @@ module.exports = function(compiler, options) {
 			return goNext();
 		}
 
-		var filename = getFilenameFromUrl(options.publicPath, compiler.outputPath, req.url);
+		var filename = getFilenameFromUrl(context.options.publicPath, context.compiler.outputPath, req.url);
 		if(filename === false) return goNext();
 
 		// in lazy mode, rebuild on bundle request
-		if(options.lazy && (!options.filename || options.filename.test(filename)))
-			rebuild();
+		if(context.options.lazy && (!context.options.filename || context.options.filename.test(filename)))
+			shared.rebuild();
 
 		if(HASH_REGEXP.test(filename)) {
```

Comment:
Could this be moved to the shared part completely? I feel like this should not be exposed.